src/App/Controls/PasswordStrengthProgressBar/PasswordStrengthViewModel.cs

```diff
@@ -0,0 +1,108 @@
+        }
+
+        public string PasswordStatus
+        {
+            get => _passwordStatus;
+            set => SetProperty(ref _passwordStatus, value);
+        }
+
+        public PasswordStrengthCategory PasswordStrengthCategory { get; set; }
+
+        public void CalculateMasterPasswordStrength(string password, string email)
+        {
+            var passwordStrength = _passwordGenerationService.PasswordStrength(password, GetPasswordStrengthUserInput(email));
+            if (string.IsNullOrEmpty(password))
+            {
+                PasswordStrength = 0;
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("DangerColor");
+                PasswordStatus = " ";
+                PasswordStrengthCategory = PasswordStrengthCategory.Weak;
+                return;
+            }
+
+            PasswordStrength = (passwordStrength.Score + 1f) / 5f;
+            if (PasswordStrength <= 0.2f)
+            {
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("DangerColor");
+                PasswordStatus = Resources.AppResources.Weak;
+                PasswordStrengthCategory = PasswordStrengthCategory.Weak;
+            }
+            else if (PasswordStrength <= 0.4f)
+            {
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("DangerColor");
+                PasswordStatus = Resources.AppResources.Weak;
+                PasswordStrengthCategory = PasswordStrengthCategory.Weak;
+            }
+            else if (PasswordStrength <= 0.6f)
+            {
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("WarningColor");
+                PasswordStatus = Resources.AppResources.Weak;
+                PasswordStrengthCategory = PasswordStrengthCategory.Weak;
+            }
+            else if (PasswordStrength <= 0.8f)
+            {
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("PrimaryColor");
+                PasswordStatus = Resources.AppResources.Good;
+                PasswordStrengthCategory = PasswordStrengthCategory.Good;
+            }
+            else if (PasswordStrength <= 1f)
+            {
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("SuccessColor");
+                PasswordStatus = Resources.AppResources.Strong;
+                PasswordStrengthCategory = PasswordStrengthCategory.Strong;
+            }
+        }
+
+        private List<string> GetPasswordStrengthUserInput(string email)
+        {
+            List<string> userInput = null;
+            var atPosition = email?.IndexOf('@');
+            if (atPosition != null && atPosition > -1)
+            {
+                var rx = new Regex("/[^A-Za-z0-9]/", RegexOptions.Compiled);
+                var data = rx.Split(email.Substring(0, atPosition.Value).Trim().ToLower());
+                userInput = new List<string>(data);
+            }
+            return userInput;
+        }
+    }
```

Comment:
I'll create a helper to have the strength calc.

Reply:
Actually i created another method on `IPasswordGenerationService` to calculate the inputs and strength in one go.
```
Result PasswordStrength(string password, List<string> userInputs = null);
Result PasswordStrength(string password, string email);
```

Reply:
Maybe that can be done but IMO an email could be a part of other user inputs, so a separate method makes more sense to increase reusability. Btw: check [this comment below](https://github.com/bitwarden/mobile/pull/2238#discussion_r1048811663) first because there are some changes that can impact here.